model/trigger/version.go

```diff
@@ -145,3 +123,52 @@ func (t *versionTriggers) generate(sub *event.Subscription) (*notification.Notif
 
 	return notification.New(t.event, sub.Trigger, &sub.Subscriber, payload)
 }
+
+func (t *versionTriggers) versionOutcome(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionSucceeded && t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *versionTriggers) versionFailure(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *versionTriggers) versionSuccess(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionSucceeded {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *versionTriggers) versionRegression(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+
```

Comment:
Side note, can't we get a logically equivalent expression by fetch the last complete version (i.e. status success or fail), and just compare it to that?

Reply:
That's different, because if this version fails due to a different task failing than the previous one we want to alert (but can't tell that from just comparing the version)

Reply:
Got it!